sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -1586,6 +1591,30 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
       }
     }
   }
+
+  test("SPARK-34726: Fix collectToPython timeouts") {
+    val listener = new QueryExecutionListener {
+      override def onFailure(funcName: String, qe: QueryExecution, exception: Exception): Unit = {}
+
+      override def onSuccess(funcName: String, qe: QueryExecution, duration: Long): Unit = {
+        // Longer than 15s in `PythonServer.setupOneConnectionServer`
```

Comment:
Let's extract the 15000 into a `private[spark] var` as member:

https://github.com/apache/spark/blob/6b18cc752379387dd13c009490b54be823cf2437/core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala#L899

And add a comment  above `// visible for testing`.
Then we can speed up this single test.

Reply:
Ok, added in https://github.com/apache/spark/commit/6b18cc752379387dd13c009490b54be823cf2437 and https://github.com/apache/spark/pull/31818/commits/8f6b811a28d1813d72151fe9063d5794b99fa481